### PR TITLE
WiiSave: Use new filesystem interface

### PR DESCRIPTION
**Pull request by leoetlino**
Migrates WiiSave to the new filesystem interface.

The first commit is straightforward: it just replaces all of the FileUtil/File calls with IOS/FS in NandStorage.

The second commit makes WiiSave use the file permission information that is returned by FS or stored in the save header. Two functions, GetFsMode and GetBinMode, were added to convert the permission values between the two formats. Files will now be saved and restored with non-hardcoded, correct values.

### CODE REVIEW

File: Source/Core/Core/HW/WiiSave.cpp [hunk 1]
```diff
@@ -204,8 +205,9 @@ class NandStorage final : public Storage
       return false;
 
     const std::string banner_file_path = m_data_dir + "/banner.bin";
-    const auto file = m_fs->CreateAndOpenFile(*m_uid, *m_gid, banner_file_path, FS::Mode::ReadWrite,
-                                              FS::Mode::ReadWrite, FS::Mode::ReadWrite);
+    const std::tuple<FS::Mode, FS::Mode, FS::Mode> modes = GetFsMode(header.permissions);
```
Comment (BhaaLseN) [inline Source/Core/Core/HW/WiiSave.cpp]: I'd really want structured bindings for this :/

**Comment by leoetlino** [comment]
Hmm, adding a `Modes` struct sounds like a good idea, as this is not the only place where we take a triplet of modes. Changing functions to take just a Modes could cut down on argument repetition. Thanks for the suggestion, I'll do this soon

**Comment by BhaaLseN** [comment]
If it plays nice with the `constexpr` there, sure, go for it!

**Comment by leoetlino** [comment]
@BhaaLseN: pushed another commit with the Modes struct change

**Pull request merged**